### PR TITLE
Add Scala Native suppresion of unused deps

### PR DESCRIPTION
**Pull request by ybasket**
Similar to #129, just for Scala Native this time. Had to add the corresponding sbt plugin as sbt-typelevel doesn't bring (unlike it does for JS), hope this doesn't have any negative consequences.

Popped up when trying to build something with native support in http4s-fs2-data.

### CODE REVIEW

File: core/build.sbt [hunk 1]
```diff
@@ -3,3 +3,4 @@ addSbtPlugin("org.typelevel" % "sbt-typelevel" % sbtTypelevelVersion)
 addSbtPlugin("org.typelevel" % "sbt-typelevel-site" % sbtTypelevelVersion)
 addSbtPlugin("org.typelevel" % "sbt-typelevel-scalafix" % sbtTypelevelVersion)
 addSbtPlugin("com.github.cb372" % "sbt-explicit-dependencies" % "0.2.16")
+addSbtPlugin("org.portable-scala" % "sbt-scala-native-crossproject" % "1.2.0")
```
Comment (armanbilge) [inline core/build.sbt:6]:
```suggestion
addSbtPlugin("org.scala-native" % "sbt-scala-native" % "0.4.0")
```

**Review by armanbilge — COMMENTED** [review]
This has been annoying me as well.

Historically Scala Native has [provoked more concern than enthusiasm](https://github.com/typelevel/sbt-typelevel/pull/43#discussion_r777824184), so I've just been inlining a fix per-project.

https://github.com/http4s/http4s-scalatags/blob/38fced20ad98c3cd40cf27844b4036bcfa2cb2e6/build.sbt#L33

**Comment by ybasket** [comment]
Thank you for the context and the fix. As for the concerns: I can also list all dependencies, but given that Scala Native gains traction in Typelevel & http4s, I would hope we can fix this for all.

**Comment by rossabaker** [comment]
If that plugin doesn't cause a problem on the modules that will never support native, centralizing the fix here seems good.

File: build.sbt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import explicitdeps.ExplicitDepsPlugin.autoImport.moduleFilterRemoveValue
+
```
Comment (armanbilge) [inline build.sbt:2]:
```suggestion
```

**Review by armanbilge — APPROVED** [review]

**Pull request merged**